DiskSetup.sh

```diff
@@ -46,14 +47,15 @@ function CarveLVM() {
    dd if=/dev/zero of="${CHROOTDEV}" bs=512 count=1000 > /dev/null 2>&1
 
    # Lay down the base partitions
-   parted -s "${CHROOTDEV}" -- mklabel msdos mkpart primary ext4 2048s ${BOOTDEVSZ} \
-      mkpart primary ext4 ${BOOTDEVSZ} 100% set 2 lvm
+   parted -s "${CHROOTDEV}" -- mklabel msdos mkpart primary "${FSTYPE}" 2048s ${BOOTDEVSZ} \
+      mkpart primary "${FSTYPE}" ${BOOTDEVSZ} 100% set 2 lvm
 
    # Stop/umount boot device, in case parted/udev/systemd managed to remount it
   systemctl stop boot.mount || true
 
    # Create LVM objects
    LVCSTAT=0
+   pvcreate "${CHROOTDEV}${PARTPRE}2" || LogBrk 5 "PV creation failed. Aborting!"
```

Comment:
We might need to revert this, or at least investigate the env in depth. The CentOS7 spel builds are hanging indefinitely on this line. RHEL7 for whatever reason is fine.

```
13:45:31 minimal-centos-7-hvm: + pvcreate /dev/xvda2
13:45:31 minimal-centos-7-hvm: File descriptor 10 (/dev/pts/0) leaked on pvcreate invocation. Parent PID 25997: bash
13:45:31 minimal-centos-7-hvm: File descriptor 11 (/dev/pts/0) leaked on pvcreate invocation. Parent PID 25997: bash
13:45:31 minimal-centos-7-hvm: File descriptor 12 (/dev/pts/0) leaked on pvcreate invocation. Parent PID 25997: bash 
```

Reply:
I imagine the difference in observed behavior is that the starting partition tables are different on the RHEL bootstrap AMI than on the CentOS bootstrap AMI. The former is using GPT; the latter (if starting from our "recovery" AMI) is using a MBR. Might try testing against the CentOS.Org AMI to be sure.

If such is the case, the problem is likely that the partition-table isn't getting sufficiently cleaned out prior to laying down new partitio-structs.